src/main/java/pl/javastart/jpaoptimalization/country/CountryService.java

```diff
@@ -1,19 +1,61 @@
 package pl.javastart.jpaoptimalization.country;
 
 import org.springframework.stereotype.Service;
+import pl.javastart.jpaoptimalization.countrylanguage.CountryLanguage;
+import pl.javastart.jpaoptimalization.countrylanguage.CountryLanguageRepository;
 
-import java.util.List;
+import java.util.*;
 
 @Service
 public class CountryService {
 
     private final CountryRepository countryRepository;
+    private final CountryLanguageRepository countryLanguageRepository;
 
-    public CountryService(CountryRepository countryRepository) {
+    public CountryService(CountryRepository countryRepository, CountryLanguageRepository countryLanguageRepository) {
         this.countryRepository = countryRepository;
+        this.countryLanguageRepository = countryLanguageRepository;
     }
 
-    public List<Country> findAll() {
-        return countryRepository.findAll();
+    public List<Country> findCountryWithLanguage() {
+        return countryRepository.findByCountryWithLanguage();
     }
+
+    public List<Country> findSorted() {
+        return countryRepository.findByOrderByName();
+    }
+
+    public Map<String, List<String>> getLanguageWithCountry() {
+        Map<String, String> countryCodes = getCodeWithCountry();
+        List<CountryLanguage> countryLanguages = countryLanguageRepository.findAll();
+        Map<String, List<String>> languageWithCountry = new TreeMap<>();
+        List<String> countries;
+        for (CountryLanguage countryLanguage : countryLanguages) {
+            String language = countryLanguage.getLanguage();
+            if (languageWithCountry.containsKey(language)) {
+                languageWithCountry.get(language).add(countryCodes.get(countryLanguage.getCountryCode()));
+            } else {
+                countries = new ArrayList<>();
+                String countryCode = countryLanguage.getCountryCode();
+                countries.add(countryCodes.get(countryCode));
+                languageWithCountry.put(language, countries);
+            }
```

Comment:
Tutaj warto użyć metody merge, pokazane w code review 
https://nurkiewicz.com/2019/03/mapmerge-one-method-to-rule-them-all.html